geode-protobuf/src/main/java/org/apache/geode/internal/protocol/protobuf/v1/operations/ExecuteFunctionOnMemberRequestOperationHandler.java

```diff
@@ -0,0 +1,147 @@
+import org.apache.geode.security.NotAuthorizedException;
+
+public class ExecuteFunctionOnMemberRequestOperationHandler implements
+    ProtobufOperationHandler<FunctionAPI.ExecuteFunctionOnMemberRequest, FunctionAPI.ExecuteFunctionOnMemberResponse> {
+  @Override
+  public Result<FunctionAPI.ExecuteFunctionOnMemberResponse, ClientProtocol.ErrorResponse> process(
+      ProtobufSerializationService serializationService,
+      FunctionAPI.ExecuteFunctionOnMemberRequest request,
+      MessageExecutionContext messageExecutionContext)
+      throws InvalidExecutionContextException, ConnectionStateException {
+
+    final String functionID = request.getFunctionID();
+    ProtocolStringList memberNameList = request.getMemberNameList();
+
+    final Function<?> function = FunctionService.getFunction(functionID);
+    if (function == null) {
+      return Failure.of(ClientProtocol.ErrorResponse.newBuilder()
+          .setError(BasicTypes.Error.newBuilder().setErrorCode(BasicTypes.ErrorCode.INVALID_REQUEST)
+              .setMessage(LocalizedStrings.ExecuteFunction_FUNCTION_NAMED_0_IS_NOT_REGISTERED
+                  .toLocalizedString(functionID))
+              .build())
+          .build());
+    }
+
+    final SecurityService securityService = messageExecutionContext.getCache().getSecurityService();
+
+    try {
+      // check security for function.
+      final String noRegion = null;
+      function.getRequiredPermissions(noRegion).forEach(securityService::authorize);
+    } catch (NotAuthorizedException ex) {
+      return Failure.of(ClientProtocol.ErrorResponse.newBuilder()
+          .setError(BasicTypes.Error.newBuilder()
+              .setMessage("Authorization failed for function \"" + functionID + "\"")
+              .setErrorCode(BasicTypes.ErrorCode.AUTHORIZATION_FAILED))
+          .build());
+    }
+
+    Set<DistributedMember> memberIds = new HashSet<>(memberNameList.size());
+    DistributionManager distributionManager =
+        messageExecutionContext.getCache().getDistributionManager();
+    for (String name : memberNameList) {
+      DistributedMember member = distributionManager.getMemberWithName(name);
+      if (member == null) {
+        return Failure.of(ClientProtocol.ErrorResponse.newBuilder()
+            .setError(BasicTypes.Error.newBuilder()
+                .setMessage("Member " + name + " not found to execute \"" + functionID + "\"")
+                .setErrorCode(BasicTypes.ErrorCode.NO_AVAILABLE_SERVER))
+            .build());
+      }
+      memberIds.add(member);
+    }
+
+    if (memberIds.isEmpty()) {
+      return Failure.of(ClientProtocol.ErrorResponse.newBuilder()
+          .setError(BasicTypes.Error.newBuilder()
+              .setMessage("No members found to execute \"" + functionID + "\"")
+              .setErrorCode(BasicTypes.ErrorCode.NO_AVAILABLE_SERVER))
+          .build());
+    }
+
+    try {
+      Execution execution;
+      if (memberIds.size() == 1) {
```

Comment:
Why do we need to check this rather than just passing an iterator with one member?

Reply:
I don't understand this comment.  The FunctionService API has two onMember methods, neither of which takes an iterator as an argument.

Reply:
I actually tried making the OnRegion integration test parameterized so it could test both onRegion and OnMember but there were too many places that required specific request/response objects.

Reply:
Oh, I'd misread the call. I think we can do the same call to `onMembers` with a Set of size 1. Under the hood, it ends up resolving to the same call.